pandas/tseries/period.py

```diff
@@ -811,6 +811,24 @@ def _format_native_types(self, na_rep=u('NaT'), date_format=None, **kwargs):
         values[imask] = np.array([formatter(dt) for dt in values[imask]])
         return values
 
+    def putmask(self, mask, value):
+        """
+        return a new Index of the values set with the mask
+
+        See also
+        --------
+        numpy.ndarray.putmask
+        """
+        if isinstance(value, Period):
```

Comment:
can you just:
```
value = Period(value)
return super(PeriodIndex, self).putmask(mask, Period(value))
```

e.g. its either the correct freq or the ``Period(value)`` will already raise
or its not a ``Period`` in which case it will raise.